lib/pluginJsonapi.js

```diff
@@ -158,18 +194,38 @@ function pluginJsonapi( server, options, next ) {
           _.each( resourceArray, function( resource ) {
             // build a href for the resource if there's an id property
             if ( resource.id ) {
-              resource.href = _buildHref( request.route.settings.bind.resourceName, resource.id );
+              resource.href = _buildHref( { type: request.route.settings.bind.resourceName, ids: resource.id } );
             }
+
             // If there's no links then there's no actions to take
             if ( resource.links ) {
+
               // Loop each resource in the links and where we have a 'type' and 'id', build a href
               _.each( resource.links, function( value ) {
+                var hrefParameters = {
+                  type: value.type
+                };
+
                 // Need both type and ids to do anything meaningful here
-                if ( value.type && _.size( value.ids ) ) {
+                if ( value.type && ( _.size( value.ids ) || _.size( value.filter ) ) ) {
+
+                  if ( _.size( value.ids ) ) {
```

Comment:
not too sure I like the use of _.size here.

Would !_.isEmpty work in its place?